configure.ac

```diff
@@ -2178,7 +2178,6 @@ AC_MSG_NOTICE([X-Accelerator-Vary support enabled: $enable_x_accelerator_vary])
 
 SQUID_AUTO_LIB(cppunit,[cppunit test framework],[LIBCPPUNIT])
 SQUID_CHECK_LIB_WORKS(cppunit,[
-  LIBCPPUNIT_LIBS="$LIBCPPUNIT_PATH -lcppunit"
   PKG_CHECK_MODULES([LIBCPPUNIT],[cppunit],[
```

Comment:
```
./configure --help
...
  LIBCPPUNIT_LIBS
              linker flags for LIBCPPUNIT, overriding pkg-config
```

Our PKG_CHECK_MODULES() promises that LIBCPPUNIT_LIBS will overwrite pkgconfig. AFAICT, this PR fixes configure.ac code to actually honor that promise. If that is true, please adjust PR description to reflect that fix.

Reply:
As per [my previous comment](https://github.com/squid-cache/squid/pull/1695#discussion_r1499295279), I think there is followup work to do

Reply:
> Alex: please adjust PR description to reflect that fix

> Francesco: I think there is followup work to do

Sure, but this change request is not about that followup work. It is about adjusting PR description to reflect a specific bug fix. I do not insist on that change, but I do not want it overlooked/ignored either.